### ISSUE TITLE
Error after `rvm get 1.26.9`

### ISSUE THREAD

**Issue by kronn** [wayneeseguin/rvm#3262]
I tried to update RVM from 1.25.15 to 1.26.9.

```
root@example.net # /usr/local/rvm/bin/rvm get 1.26.9

Hi there, it looks like you have requested updating rvm to version 1.26.9,
if your intention was ruby installation use instead: rvm install 1.26.9

Downloading https://get.rvm.io
Downloading https://github.com/wayneeseguin/rvm/archive/1.26.9.tar.gz
Downloading https://github.com/wayneeseguin/rvm/releases/download/1.26.9/1.26.9.tar.gz.asc
gpg: Signature made Mi 07 Jan 2015 12:55:10 CET using RSA key ID BF04FF17
gpg: Good signature from "Michal Papis (RVM signing) <mpapis@gmail.com>"
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 409B 6B17 96C2 7546 2A17  0311 3804 BB82 D39D C0E3
     Subkey fingerprint: 62C9 E5F4 DA30 0D94 AC36  166B E206 C29F BF04 FF17
GPG verified '/usr/local/rvm/archives/rvm-1.26.9.tgz'

Upgrading the RVM installation in /usr/local/rvm/
Upgrade of RVM in /usr/local/rvm/ is complete.

# xunde,
#
#   Thank you for using RVM!
#   We sincerely hope that RVM helps to make your life easier and more enjoyable!!!
#
# ~Wayne, Michal & team.

In case of problems: http://rvm.io/help and https://twitter.com/rvm_io

Upgrade Notes:

  * No new notes to display.

/usr/local/rvm/bin/rvm: line 67: unexpected EOF while looking for matching `"'
/usr/local/rvm/bin/rvm: line 70: syntax error: unexpected end of file
/usr/local/rvm/bin/rvm: line 1: return: can only `return' from a function or sourced script
```

The exitstatus is "2".

However, the new version is installed.

Sadly, our automation setup only sees the exitstatus and fails.

The new rvm-script has fewer lines than noted in the error message. therefore, I have problems debugging this myself.

**Comment by kronn** [comment]
The username "xunde" is the name of the login-user

```
root@example.net # env | grep USER
USER=root
SUDO_USER=xunde
USERNAME=root
```

**Comment by kronn** [comment]
I just checked on another machine having the same issue.

The previous /usr/local/rvm/bin/rvm only has 61 lines.

Can you explain that error?

**Comment by mpapis** [comment]
does the error happen again with the new version? - if it does please run:
```bash
/usr/local/rvm/bin/rvm --trace get 1.26.9 2>&1 | tee big.log
```
and [gist](https://gist.github.com) `big.log`.

**Comment by kronn** [comment]
With the new version, the script only ends with `/usr/local/rvm/bin/rvm: line 1: return: can only `return' from a function or sourced script` and returncode 0.

The big-log has been gisted: https://gist.github.com/kronn/53f5ce82c76dc006fe55

**Comment by xeron** [comment]
Same issue with rvm 1.26.10 after rvm reload:
```
xeron@Ivans-MacBook-Pro:~$ rvm reload
RVM reloaded!
xeron@Ivans-MacBook-Pro:~$ logout
-bash: return: can only `return' from a function or sourced script

[Process completed]
```

**Comment by raffomania** [comment]
Just installed rvm, got the same problem - Note, I'm using the fish wrapper:

    $ rvm reload
    RVM reloaded!
    /tmp/rvm.fish.c5uoq9ooP3: line 1: return: can only `return' from a function or sourced script

**Comment by bryanroth** [comment]
I was having the same problem but adding the following line to my `~/.bashrc` file resolved the issue:

    [[ -s "$HOME/.rvm/src/rvm/scripts/rvm" ]] && . "$HOME/.rvm/src/rvm/scripts/rvm"

**Comment by xeron** [comment]
I have:
```
export PATH=$PATH:$HOME/.rvm/bin
[[ -s "$HOME/.rvm/scripts/rvm" ]] && source "$HOME/.rvm/scripts/rvm"
[[ -r $rvm_path/scripts/completion ]] && . $rvm_path/scripts/completion
```
which should be enough.

**Comment by killphi** [comment]
Hi,

I wanted to chime in and say that I have the same problem, also using the fish wrapper.

Actually, I dislike doing that without having something to contribute other than "Hey, I want that fixed."
So I traced the error message back to https://github.com/wayneeseguin/rvm/blob/master/scripts/functions/environment#L207

It seems like my `bash --version`
```
GNU bash, version 4.3.33(1)-release (x86_64-redhat-linux-gnu)
```
does not like to handle `return` statements inside traps.
This is the output I got from `bash -xv` and a SIGHUP:
```bash
[snip]
++ return 0
++ set +x
bash: return: can only `return' from a function or sourced script
status=$? ; __rvm_teardown_final ; set +x ; return $status
bash: return: can only `return' from a function or sourced script
```

Changing the `return` to `exit` fixed the EXIT trap for me.
However, since my experience in that field is limited to non-existent, I am not sure whether therein lies the solution.

My system is running an up to date Fedora 21.

PS:
@bryanroth & @xeron: neither of your solutions worked, but thanks for trying to help :)

**Comment by mpapis** [comment]
@killphi please open a pull request removing the `status` and `return`, it looks like over-engineered.

**Comment by xeron** [comment]
@killphi sorry I haven't meant it fixes a problem. I have such code in my bash profile and it still complains about return.

**Comment by mpapis** [comment]
fixed in #3340
